### ISSUE TITLE
Create a GPU debug print

### ISSUE THREAD

**Issue by Lut99** [Lut99/RayTracer-2#22]
While developing on the GPU, the one thing that is missing is a way to print to stderr (only stdout if possible by default). That is why we will create a library which can be used to print to stderr from a GPU, in a C++-likable way (with stream operators, that is). Preferably, this method should be an ostream-derived class so that classes only need to implement a single write function, but with the __host__ __device__ stuff, this might not be feasable. Instead, we will simply make a standalone ostream-like class.

**Comment by Lut99** [comment]
Closed for now because the Debug printing from the GPU is hard because no general synchronization is present (there is block-level synchronization, so maybe we can do something with that?). Additionally, looked at making a Debug suite for CPU as well which applies stacktracing in case of signals (e.g., segfault :) ). Seems to work decently for single threads, but for multiple threads it will race condition for sure. Additionally, making a GPU-side equivalent of this also quite hard, especially because I don't now if signals are occuring on the GPU.

Long story short, cool idea, useful, but not really worth the time right now I reckon.
